src/main/java/com/slimebot/level/Level.java

```diff
@@ -0,0 +1,130 @@
+package com.slimebot.level;
+
+import com.slimebot.main.Main;
+import com.slimebot.main.config.guild.GuildConfig;
+import com.slimebot.util.MathUtil;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.channel.concrete.TextChannel;
+import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.jetbrains.annotations.NotNull;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public record Level(long guildId, long userId, int level, int xp) implements Comparable<Level> {
+
+    public void save() {
+        Main.database.run(handle -> handle.createUpdate("insert into levels values(:guild, :user, :level, :xp) on conflict(guild, \"user\") do update set level = :level, xp = :xp")
+                .bind("guild", guildId())
+                .bind("user", userId())
+                .bind("level", level())
+                .bind("xp", xp())
+                .execute());
+    }
+
+    public Level add(int level, int xp) {
+        return new Level(guildId(), userId(), level()+level, xp()+xp);
+    }
+
+    @Override
+    public int compareTo(@NotNull Level o) {
+        int levelComp = Integer.compare(o.level, this.level);
+        return levelComp != 0 ? levelComp : Double.compare(o.xp, this.xp);
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Why do you implement toString()? I think that's not required

Reply:
for debug prints. since it doesn't interfere I would leave it in to make debugging with the level system easier in the future